src/cdk/datetime/datepicker.spec.ts

```diff
@@ -0,0 +1,646 @@
+import {
+  dispatchFakeEvent,
+  DEC,
+  JAN,
+  JUL,
+  JUN,
+} from '@angular/cdk/testing';
+import {NativeDateModule} from '@angular/cdk/datetime';
+import {Component, FactoryProvider, Type, ValueProvider, ViewChild} from '@angular/core';
+import {ComponentFixture, fakeAsync, flush, TestBed} from '@angular/core/testing';
+import {FormControl, FormsModule, ReactiveFormsModule} from '@angular/forms';
+import {By} from '@angular/platform-browser';
+import {NoopAnimationsModule} from '@angular/platform-browser/animations';
+import {CdkDatepicker} from './datepicker';
+import {CdkDatepickerInput} from './datepicker-input';
+import {CdkDatepickerModule} from './index';
+import {BrowserDynamicTestingModule} from '@angular/platform-browser-dynamic/testing';
+
+describe('CdkDatepicker', () => {
+  // Creates a test component fixture.
+  function createComponent(
+      component: Type<any>,
+      imports: Type<any>[] = [],
+      providers: (FactoryProvider | ValueProvider)[] = [],
+      entryComponents: Type<any>[] = []): ComponentFixture<any> {
+
+    TestBed.configureTestingModule({
+      imports: [
+        FormsModule,
+        CdkDatepickerModule,
+        NoopAnimationsModule,
+        ReactiveFormsModule,
+        ...imports
+      ],
+      providers,
+      declarations: [component, ...entryComponents],
+    });
+
+    TestBed.overrideModule(BrowserDynamicTestingModule, {
```

Comment:
`overrideModule` should be avoided because it won't work with AoT unit tests, which we will switch to eventually